azurelinuxagent/ga/policy/policy_engine.py

```diff
@@ -15,10 +15,37 @@
 # Requires Python 2.4+ and Openssl 1.0+
 #
 
+import json
+import os
+import re
 from azurelinuxagent.common import logger
 from azurelinuxagent.common.event import WALAEventOperation, add_event
 from azurelinuxagent.common import conf
 from azurelinuxagent.common.exception import AgentError
+from azurelinuxagent.common.protocol.extensions_goal_state_from_vm_settings import _CaseFoldedDict
+
+# Schema for policy file.
+_POLICY_SCHEMA = \
+    {
+        "policyVersion": str,
+        "extensionPolicies": {
+            "allowListedExtensionsOnly": bool,
+            "signatureRequired": bool,
+            "extensions": {
+                "<extensionName>": {
+                    "signatureRequired": bool
+                }
+            }
+        }
+    }
+
+# Default policy values to be used when customer does not specify these attributes in the policy file.
+_DEFAULT_ALLOW_LISTED_EXTENSIONS_ONLY = False
+_DEFAULT_SIGNATURE_REQUIRED = False
+
+# Agent supports up to this version of the policy file ("policyVersion" in schema).
+# Increment this number when any new attributes are added to the policy schema.
+_MAX_SUPPORTED_POLICY_VERSION = "0.1.0"
```

Comment:
Do we want this to be "0.1.0" to accommodate any changes we might make before GA? Or set to "1.0.0" now?